prow/jenkins/jenkins.go

```diff
@@ -145,8 +145,8 @@ func (c *Client) Build(pj *kube.ProwJob) (*url.URL, error) {
 		return nil, err
 	}
 	defer resp.Body.Close()
-	if resp.StatusCode != 201 {
-		return nil, fmt.Errorf("response not 201: %s", resp.Status)
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return nil, fmt.Errorf("response not 2XX: %s", resp.Status)
 	}
 	return resp.Location()
```

Comment:
I believe the 200 doesn't come with this header

Reply:
Yup, that's the problem with the 302->200 response. It really should only be returned when an error prevents a new build from getting created. 

We are now investigating why this is so in jobs synchronized by the jenkins-sync-plugin. It doesn't occur with regular Jenkins jobs.